### ISSUE TITLE
Rationalize use of .vue extensions

### ISSUE THREAD

**Issue by nicksellen** [yunity/karrot-frontend#626]
Currently we have a slightly weird and inconsistent usage of .vue extensions in import statements.

## tl;dr

I would like to switch to always importing components with this syntax:

```js
import MyComponent from '@/components/MyComponent'
```

... but intellij does not support it without using the EAP version, or waiting a bit until they release 2017.3.

Question: do you use an editor that would not support this? Or have any other reason not to do it like this?

### variations

These are some of the possible usages possibilities:

```js
import MyComponent from '@/components/MyComponent'
```

```js
import MyComponent from '@/components/MyComponent.vue'
```

```js
import MyComponent from './MyComponent.vue'
```

There are 3 environments that I use that are impacted by this:

### webpack

This bit of config that we use allows webpack to import .vue files without providing the extension. All 3 import styles work fine.

```js
resolve: {
    extensions: ['.js', '.vue', '.json'],
...
```

### jest

Currently, we have this bit of config, that adds the extension to components:

```json
"moduleNameMapper": {
  "@/components/(.*)$": "<rootDir>/src/components/$1.vue",
  "@/(.*)$": "<rootDir>/src/$1",
  ...
}
```
So, `@/components/MyComponent.vue` will _not_ work (it will try and import `@/components/MyComponent.vue.vue`)

But there is another bit of config we could add:

```json
"moduleFileExtensions": ["js", "json", "node", "vue"],
```

then we could remove the `@/components` alias (as the `@` one would work).

### intellij imports

Intellij does not currently support leaving out the .vue extension, so will have a "failed to resolve module" kind of error.

Intellij 2017.3 includes support for .vue imports without the extension though [1] which will be released "in the fall" [2] - or install the EAP version now...

[1] https://intellij-support.jetbrains.com/hc/en-us/community/posts/115000556284-vue-import-component-Module-is-not-installed-?page=1#community_comment_115000480870 
[2] https://blog.jetbrains.com/idea/2017/09/whats-new-in-intellij-idea-2017-3-eap/

**Comment by tiltec** [comment]
I'm going to remove all .vue extensions now
